GitUI/FilterBranchHelper.cs

```diff
@@ -117,42 +119,15 @@ public void InitToolStripBranchFilter()
             _NO_TRANSLATE_toolStripBranches.Enabled = Module.IsValidGitWorkingDir();
         }
 
-        private List<string> GetBranchHeads(bool local, bool remote)
-        {
-            var list = new List<string>();
-            if (local && remote)
-            {
-                var branches = Module.GetRefs(true, true);
-                list.AddRange(branches.Where(branch => !branch.IsTag).Select(branch => branch.Name));
-            }
-            else if (local)
-            {
-                var branches = Module.GetRefs(false);
-                list.AddRange(branches.Select(branch => branch.Name));
-            }
-            else if (remote)
-            {
-                var branches = Module.GetRefs(true, true);
-                list.AddRange(branches.Where(branch => branch.IsRemote && !branch.IsTag).Select(branch => branch.Name));
-            }
-
-            return list;
-        }
-
-        private IEnumerable<string> GetTagsRefs()
-        {
-            return Module.GetRefs(true, false).Select(tag => tag.Name);
-        }
-
         private List<string> GetBranchAndTagRefs(bool local, bool tag, bool remote)
         {
-            var list = GetBranchHeads(local, remote);
-            if (tag)
-            {
-                list.AddRange(GetTagsRefs());
-            }
+            // Options are interpreted as the refs the search should be limited too
```

Comment:
This is a scope creep compared to master, I opted to add a comment instead of code to limit the functionality